### DISCUSSION TITLE
Can `prompt.ask` be cancelled using escape?

### DISCUSSION THREAD

**Discussion by fiedl** [piotrmurach/tty-prompt#170] [Q&A] [answered]
I'm asking a simple question like this:

```ruby
name = prompt.ask "What is your name?", default: "John Doe"
```

Can I somehow have the `prompt.ask` question be cancelled when the user presses the <kbd>ESC</kbd> key and the code would resume?

I've been experimenting with the `keyescape` event, but so far unsuccessful.

```ruby
prompt.on(:keyescape) { print "escape has been pressed."; @escape_pressed = true }
name = prompt.ask "What is your name?", default: "John Doe"
unless @escape_pressed
  # ...
end
```

In the above code, the `keyescape` event is triggered as expected while waiting for the user to answer the question. But I didn't know how to cancel the question from the event handler.

Thank you very much!

**Comment by piotrmurach** [comment, marked as answer]
You could raise an error:

```ruby
prompt.on(:keypress) do
  raise EscapePressedError
end
```

Then rescue an `ask` call like so:

```ruby
begin
  name = prompt.ask "What is your name?", default: "John Doe"
rescue EscapePressedError
  puts "bailing out..."
end
```

**Reply by fiedl** [reply]
Thank you for the quick reply, this works! 🎉
